builtin/providers/aws/resource_aws_db_option_group.go

```diff
@@ -0,0 +1,288 @@
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsDbOptionGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDbOptionGroupCreate,
+		Read:   resourceAwsDbOptionGroupRead,
+		Update: resourceAwsDbOptionGroupUpdate,
+		Delete: resourceAwsDbOptionGroupDelete,
+		Schema: map[string]*schema.Schema{
+			"arn": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"option_group_name": &schema.Schema{
+				Type:         schema.TypeString,
+				ForceNew:     true,
+				Required:     true,
+				ValidateFunc: validateDbOptionGroupName,
+			},
+			"engine_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"major_engine_version": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"option_group_description": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"option": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"option_name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"port": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+						},
+						"db_security_group_memberships": &schema.Schema{
+							Type:     schema.TypeSet,
+							Optional: true,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Set:      schema.HashString,
+						},
+						"vpc_security_group_memberships": &schema.Schema{
+							Type:     schema.TypeSet,
+							Optional: true,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Set:      schema.HashString,
+						},
+					},
+				},
+				Set: resourceAwsDbOptionHash,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsDbOptionGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	rdsconn := meta.(*AWSClient).rdsconn
+	tags := tagsFromMapRDS(d.Get("tags").(map[string]interface{}))
+
+	createOpts := &rds.CreateOptionGroupInput{
+		EngineName:             aws.String(d.Get("engine_name").(string)),
+		MajorEngineVersion:     aws.String(d.Get("major_engine_version").(string)),
+		OptionGroupDescription: aws.String(d.Get("option_group_description").(string)),
+		OptionGroupName:        aws.String(d.Get("option_group_name").(string)),
+		Tags:                   tags,
+	}
+
+	log.Printf("[DEBUG] Create DB Option Group: %#v", createOpts)
+	_, err := rdsconn.CreateOptionGroup(createOpts)
+	if err != nil {
+		return fmt.Errorf("Error creating DB Option Group: %s", err)
+	}
+
+	d.SetId(d.Get("option_group_name").(string))
+	log.Printf("[INFO] DB Option Group ID: %s", d.Id())
+
+	return resourceAwsDbOptionGroupRead(d, meta)
```

Comment:
Should we call `resourceAwsDbOptionGroupUpdate` here instead, to apply the options?